src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py

```diff
@@ -2245,7 +2245,64 @@ def _reindex_axis_0(
         modin_frame = self._modin_frame
         method = kwargs.get("method", None)
         fill_value = kwargs.get("fill_value", np.nan)  # type: ignore[arg-type]
+        limit = kwargs.get("limit", None)
         _filter_column_snowflake_quoted_id = None
+        is_index = kwargs.get("_is_index", False)
+        if is_index:
+            modin_frame = modin_frame.ensure_row_position_column()
+            row_position_column = modin_frame.row_position_snowflake_quoted_identifier
+            modin_frame = InternalFrame.create(
```

Comment:
can't we use `append_column` to add the row position column as well?

Reply:
append_column will add a duplicate physical row order column - I'd rather not add the additional column to the physical object.